src/registers/debug.rs

```diff
@@ -0,0 +1,599 @@
+        lsb..lsb + 2
+    }
+}
+
+impl TryFrom<usize> for HwBreakpointSize {
+    type Error = TryFromIntError;
+
+    fn try_from(size: usize) -> Result<Self, Self::Error> {
+        Self::new(size).ok_or(TryFromIntError(()))
+    }
+}
+
+/// A valid value of the [`Dr7`] debug register.
+///
+/// In addition to the [`Dr7Flags`] this value has a condition field and a size field for each debug address register.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[repr(transparent)]
+pub struct Dr7Value {
+    bits: u64,
+}
+
+impl From<Dr7Flags> for Dr7Value {
+    fn from(dr7_flags: Dr7Flags) -> Self {
+        Self::from_bits_truncate(dr7_flags.bits())
+    }
+}
+
+impl Dr7Value {
+    const fn mask() -> u64 {
+        let field_mask = (1 << 32) - (1 << 16);
+        let flag_mask = Dr7Flags::all().bits();
+        field_mask | flag_mask
+    }
+
+    /// Convert from underlying bit representation, unless that representation contains bits that do not correspond to a field.
+    #[inline]
+    pub const fn from_bits(bits: u64) -> Option<Self> {
+        if (bits & Self::mask()) == 0 {
+            Some(Self { bits })
+        } else {
+            None
+        }
+    }
+
+    /// Convert from underlying bit representation, dropping any bits that do not correspond to fields.
+    #[inline]
+    pub const fn from_bits_truncate(bits: u64) -> Self {
+        Self {
+            bits: bits & !Self::mask(),
+        }
+    }
+
+    /// Convert from underlying bit representation, preserving all bits (even those not corresponding to a defined field).
+    ///
+    /// # Safety
+    ///
+    /// The bit representation must be a valid [`Dr7Value`].
+    #[inline]
+    pub const unsafe fn from_bits_unchecked(bits: u64) -> Self {
+        Self { bits }
+    }
+
+    /// Returns the raw value of the fields currently stored.
+    #[inline]
+    pub const fn bits(&self) -> u64 {
+        self.bits
+    }
+
+    /// Returns the [`Dr7Flags`] in this value.
+    #[inline]
+    pub const fn flags(self) -> Dr7Flags {
+        // SAFETY: The extra bits used in fields are valid in our representation of `Dr7Flags`.
+        unsafe { Dr7Flags::from_bits_unchecked(self.bits) }
+    }
+
+    /// Returns a mutable reference to the [`Dr7Flags`] in this value.
+    ///
+    /// This can be used to modify the flags of this value in-place.
+    #[inline]
+    pub fn flags_mut(&mut self) -> &mut Dr7Flags {
+        // SAFETY: The extra bits used in fields are valid in our representation of `Dr7Flags`.
+        // SAFETY: `Dr7Flags` has the same representation as `Dr7Value`.
+        unsafe { mem::transmute(self) }
+    }
+
+    /// Returns the condition field of a debug address register.
+    pub fn get_condition(&self, n: DebugAddressRegisterNumber) -> HwBreakpointCondition {
+        let condition = self.bits.get_bits(HwBreakpointCondition::bit_range(n));
+        // SAFETY: All two bit patterns are valid variants
+        unsafe { mem::transmute(condition as u8) }
```

Comment:
I'm not a fan of using `transmute` because it makes refactorings more dangerous. I would prefer a `TryFrom<u8>` implementation and then `condition.try_into().expect("condition should be always valid")`.

Reply:
I added and migrated to `from_bits` functions, as a `TryFrom` implementation would be ambiguous for `HwBreakpointSize`'s `TryFrom` implementation, which does not convert from a bit pattern but from the semantic value.